crypto/Kconfig

```diff
@@ -50,28 +50,9 @@ config CRYPTO_CRYPTODEV_HARDWARE
 	depends on CRYPTO_CRYPTODEV
 	default n
 
-config CRYPTO_SW_AES
-	bool "Software AES library"
-	depends on ALLOW_BSD_COMPONENTS
```

Comment:
I generally agree here to just let the linker deal with it, but if want to actually have this `ALLOW_BSD_COMPONENTS` flag we cannot merge this in. That said I don't think this flag provides much value and I don't think it guards all BSD code.

Reply:
Move ALLOW_BSD_COMPONENTS to config CRYPTO, since the whole crpyto driver framework come from FreeBSD.